src/main/java/at/ac/tuwien/kr/alpha/common/atoms/Literal.java

```diff
@@ -75,9 +92,8 @@
 	 * 
 	 * @return
 	 */
-	default Set<VariableTerm> getNonBindingVariables() {
+	public Set<VariableTerm> getNonBindingVariables() {
 		if (isNegated()) {
-			Atom atom = getAtom();
 			return SetUtils.union(atom.getBindingVariables(), atom.getNonBindingVariables());
 		} else {
 			return Collections.emptySet();
```

Comment:
I think it should be `atom.getNonBindingVariables()` here. Or is it guaranteed that each `Literal` has only binding variables? In that case the union in the other if-branch should be removed and replaced with just the binding variables.

Reply:
Thanks for pointing this out, I actually introduced some issues here. Now, [LiteralBindingNonBindingVariablesTest](https://github.com/alpha-asp/Alpha/blob/issue_117/src/test/java/at/ac/tuwien/kr/alpha/common/atoms/LiteralBindingNonBindingVariablesTest.java) ensures that binding and non-binding variables are the same as [before my changes](https://github.com/alpha-asp/Alpha/blob/test/non-binding_vars/src/test/java/at/ac/tuwien/kr/alpha/common/atoms/AtomBindingNonBindingVariablesTest.java) (see additional elaboration [below](https://github.com/alpha-asp/Alpha/pull/118#issuecomment-381642604)).